src/lib.rs

```diff
@@ -273,6 +275,46 @@ pub enum Scenario {
     IncomingCall,
 }
 
+#[derive(Clone)]
+pub struct Progress {
+    pub tag: String,
+    pub title: String,
+    pub status: String,
+    /// `value` should be in the range of 0.0 to 1.0.
+    pub value: f32,
+    pub value_string: String,
+}
```

Comment:
the fields should not be public to allow us to add more in the future, we should instead make a `ProgressBuilder` with methods to create `Progress`.

Reply:
Thank you for your feedback! I'm not very familiar with the best practices for writing libraries. Are you suggesting creating code like the following?

```rust
pub struct Progress {
    tag: String,
    title: String,
    // other fields...
}

impl Progress {
    fn xml() -> ‘static str {
    ...
    }
    // other methods...
}

pub struct ProgressBuilder {
    tag: String,
    title: String,
    // other fields...
}

impl ProgressBuilder {
    pub fn new() -> Self {
        ProgressBuilder {
            tag: String::new(),
            title: String::new(),
            status: String::new(),
            value: 0.0,
            value_string: String::new(),
        }
    }

    pub fn tag(mut self, tag: &str) -> Self {
        self.tag = tag.to_owned();
        self
    }

    pub fn title(mut self, title: &str) -> Self {
        self.title= title.to_owned();
        self
    }
    // other methods...

    pub fn build(self) -> Progress {
        Progress {
            tag: self.tag,
            title: self.title,
            status: self.status,
            value: self.value,
            value_string: self.value_string,
        }
    }
}
```
Alternatively, if you feel it's more appropriate to modify it yourself after the merge, I trust your judgment on the best approach. I appreciate your guidance!